airbyte-ci/connectors/regression-testing/src/regression_testing/connector_runner.py

```diff
@@ -0,0 +1,378 @@
+                "--state",
+                self.IN_CONTAINER_STATE_PATH,
+            ],
+            raise_container_error,
+            config=config,
+            catalog=catalog,
+            state=state,
+            enable_caching=enable_caching,
+        )
+
+    async def get_container_env_variable_value(self, name: str) -> str:
+        return await self._connector_under_test_container.env_variable(name)
+
+    async def get_container_label(self, label: str):
+        return await self._connector_under_test_container.label(label)
+
+    async def get_container_entrypoint(self):
+        entrypoint = await self._connector_under_test_container.entrypoint()
+        return " ".join(entrypoint)
+
+    async def _run(
+        self,
+        airbyte_command: List[str],
+        raise_container_error: bool,
+        config: SecretDict = None,
+        catalog: dict = None,
+        state: Union[dict, list] = None,
+        enable_caching=True,
+    ) -> List[AirbyteMessage]:
+        """Run a command in the connector container and return the list of AirbyteMessages emitted by the connector.
+
+        Args:
+            airbyte_command (List[str]): The command to run in the connector container.
+            raise_container_error (bool): Whether to raise an error if the container fails to run the command.
+            config (SecretDict, optional): The config to mount to the container. Defaults to None.
+            catalog (dict, optional): The catalog to mount to the container. Defaults to None.
+            state (Union[dict, list], optional): The state to mount to the container. Defaults to None.
+            enable_caching (bool, optional): Whether to enable command output caching. Defaults to True.
+
+        Returns:
+            List[AirbyteMessage]: The list of AirbyteMessages emitted by the connector.
+        """
+        container = self._connector_under_test_container
+        if not enable_caching:
+            container = container.with_env_variable("CAT_CACHEBUSTER", str(uuid.uuid4()))
+        if config:
+            container = container.with_new_file(self.IN_CONTAINER_CONFIG_PATH, contents=json.dumps(dict(config)))
+        if state:
+            container = container.with_new_file(self.IN_CONTAINER_STATE_PATH, contents=json.dumps(state))
+        if catalog:
+            container = container.with_new_file(self.IN_CONTAINER_CATALOG_PATH, contents=catalog.json())
+
+        await self._write_output(airbyte_command, container)
+
+    async def _write_output(self, airbyte_command: list, container: dagger.Container):
+        filename = f"raw_output.txt"
+        local_output_file_path = f"{self._output_directory}/{filename}"
+
+        entrypoint = await container.entrypoint()
+        airbyte_command = entrypoint + airbyte_command
+        container = container.with_exec(
+            ["sh", "-c", " ".join(airbyte_command) + f" > {self.IN_CONTAINER_OUTPUT_PATH} 2>&1 | tee -a {self.IN_CONTAINER_OUTPUT_PATH}"],
```

Comment:
this assumes `sh` and `tee` are available in the container.
I think we should rather await on a with exec which is not altering the original entrypoint and get stdout + stderr.
And as I suggested above, I think this can happen in a different class, maybe at the backend class level.
E.G:
`_run` "just" returns:
 ```python
return command, await container.with_exec(airbyte_command)
```

Then in a different code path:
```python
command, executed_container = connector_run.call_read(***)

stdout, stderr = await executed_container.stdout(), await executed_container.stderr()
# These stdout stderr can be written to files, db are whatever the "backend" implements
```